src/main/java/org/scijava/ui/swing/widget/SwingNumberWidget.java

```diff
@@ -119,6 +119,11 @@ else if (model.isStyle(NumberWidget.SLIDER_STYLE)) {
 		final SpinnerNumberModel spinnerModel =
 			new SpinnerNumberModelFactory().createModel(value, min, max, stepSize);
 		spinner = new JSpinner(spinnerModel);
+		String format = getFormat(model);
+		if (format != null) {
+			spinner.setEditor(new JSpinner.NumberEditor(spinner, format));
```

Comment:
It would be great to have some sensible guessing on the format (based on `min`, `max` and/or default value) in case `format` is `null` here.